ignite/contrib/handlers/lr_finder.py

```diff
@@ -0,0 +1,425 @@
+# coding: utf-8
+
+from ignite.engine import Events, Engine
+from ignite.contrib.handlers import LRScheduler
+from torch.optim.lr_scheduler import _LRScheduler
+import numpy as np
+import torch
+import copy
+import os
+import logging
+import weakref
+import warnings
+
+
+class FastaiLRFinder(object):
+    """
+    Learning rate finder handler for supervised trainers.
+
+    While attached, the handler increases the learning rate in between two
+    boundaries in a linear or exponential manner. It provides valuable
+    information on how well the network can be trained over a range of learning
+    rates and what is the optimal learning rate.
+
+    Args:
+        model (`torch.nn.Module`): the model to train.
+        optimizer (`torch.optim.Optimizer`): the optimizer to use, the defined
+        optimizer learning rate is assumed to be the lower boundary of the range
+        test.
+        num_iter (int): number of iterations for lr schedule between base lr
+            and end_lr. If `None` it will run for
+            `len(dataloader) * trainer.state.max_epochs`
+        output_transform (callable, optional): function that transform the
+            engine's state.output after each iteration. It must return the loss
+            of that iteration.
+        end_lr (float): upper bound for lr search.
+        step_mode (str): "exp" or "linear", which way should the lr be increased
+            from optimizer's initial lr to end_lr
+        smooth_f (float): loss smoothing factor in range [0, 1)
+        diverge_th (float): Used for stopping the search when
+            `current loss > diverge_th * best_loss`
+        memory_cache (bool): if this flag is set to True, `state_dict` of model
+            and optimizer will be cached in memory. Otherwise, they will be
+            saved to files under the `cache_dir`.
+        cache_dir (str): path for storing temporary files. If no path is
+            specified, system-wide temporary directory is used. Notice that this
+             parameter will be ignored if `memory_cache` is True.
+
+    Examples:
+
+        .. code-block:: python
+
+            from ignite.contrib.handlers import LRFinder
+
+            # Create a lr_finder
+            lr_finder = LRFinder(model, optimizer)
+
+            # Attach the lr_finder to the trainer
+            lr_finder.attach(trainer)
+
+            # Run trainer
+            trainer.run(dataloader)
+
+            # Detach lr_finder
+            lr_finder.detach()
```

Comment:
I think we can detach the lr_finder automatically.

Anyway, right now the user can't use the `lr_finder` multiple times with different arguments since they are all passed during `__init__`. maybe we can move some of those parameters to `attach` instead.